module-services/service-desktop/parser/MessageHandler.cpp

```diff
@@ -37,12 +38,17 @@ namespace sdesktop::endpoints
 
         auto handler = endpointFactory->create(*context, OwnerServicePtr);
 
-        if (handler != nullptr) {
-            handler->handle(*context);
-        }
-        else {
+        if (handler == nullptr) {
             LOG_ERROR("No way to handle!");
+            return;
+        }
+        handler->handle(*context);
+        auto ownerService = dynamic_cast<ServiceDesktop *>(OwnerServicePtr);
+        if (ownerService == nullptr) {
+            LOG_ERROR("Downcasting error!");
+            return;
         }
+        ownerService->restartNotificationsClearTimer();
```

Comment:
why not just bind ownerService to the ServiceDesktop? or better: pass dependencies in ctor, in this case: callback to `clear`. or: better: if timer is required by this scope: why not create it here?

Reply:
Done by passing callback